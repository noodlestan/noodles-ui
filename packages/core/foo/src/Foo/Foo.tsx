import { Component, JSX } from 'solid-js';

type Props = {
    num: number;
    children?: JSX.Element;
};

export const Foo: Component<Props> = props => {
    return (
        <div>
            {props.num}
            <hr />
            {props.children}
        </div>
    );
};
