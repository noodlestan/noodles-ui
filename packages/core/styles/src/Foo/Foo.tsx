import { Component, JSX, createSignal } from 'solid-js';

import { sum } from './util';

type Props = {
    startAt: number;
    children?: JSX.Element;
};

export const Foo: Component<Props> = props => {
    const [num, setNum] = createSignal(0);

    return (
        <div>
            {num() + props.startAt}
            <button onClick={() => setNum(v => sum(v, 1))}>+1</button>
            <hr />
            {props.children}
        </div>
    );
};
