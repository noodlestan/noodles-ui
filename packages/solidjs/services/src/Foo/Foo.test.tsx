import { fireEvent, render } from '@solidjs/testing-library';
import { describe, expect, test } from 'vitest';

import { Foo } from './Foo';

// NOTE! this file is preserved for sanity
// ts build ambient is brittle and when broken it's cool to have a minimum repro to work with
describe('<Foo />', () => {
    test('renders', () => {
        const { container, unmount } = render(() => <Foo num={4} />);
        expect(container.innerHTML).toMatchSnapshot();
        unmount();
    });

    // Note: the below test is taken straight from the official docs
    // I am sure there must be a more ergonomic way of triggering events
    test('updates', async () => {
        const { container, unmount } = render(() => <Foo num={4}>foo</Foo>);
        const buttonClicked = new Promise(resolve => {
            const handler = (ev: MouseEvent) => {
                container.removeEventListener('click', handler);
                resolve(ev);
            };
            container.addEventListener('click', handler);
        });
        fireEvent.click(container);
        await buttonClicked;
        expect(container.innerHTML).toMatchSnapshot();
        unmount();
    });
});
