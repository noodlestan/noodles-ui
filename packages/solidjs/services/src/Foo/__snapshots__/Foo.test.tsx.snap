// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Foo /> > renders 1`] = `"<div>4<hr></div>"`;

exports[`<Foo /> > updates 1`] = `"<div>4<hr>foo</div>"`;
